app/pb/search.proto

```diff
@@ -0,0 +1,115 @@
+syntax = "proto3";
+
+package org.couchers.api.search;
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/wrappers.proto";
+
+import "pb/api.proto";
+import "pb/communities.proto";
+import "pb/discussions.proto";
+import "pb/groups.proto";
+import "pb/pages.proto";
+
+service Search {
+  // This is a secure service: a user needs to be authenticated and not jailed to call functions here, refer to auth.proto and jail.proto
+
+  rpc Search(SearchReq) returns (SearchRes) {
+    // Search all content
+  }
+
+  rpc UserSearch(UserSearchReq) returns (UserSearchRes) {
+    // Search for users in particular with extra filters, e.g. hosts
+  }
+}
+
+message Area {
+  double lat = 1;
+  double lng = 2;
+  // truncated to between 5000 (5 km) and 200000 (200 km) for privacy
+  double radius = 3; // m
+}
+
+message SearchReq {
+  string query = 1;
+
+  bool include_users = 4;
+  bool include_communities = 5;
+  bool include_groups = 6;
+  bool include_places = 7;
+  bool include_guides = 8;
+  bool include_discussions = 9;
+  // bool include_events = 10;
+
+  // at most one of the following
+  Area search_in_area = 11;
+  int64 search_in_community_id = 12;
```

Comment:
Add oneof here so this is enforced?

Reply:
Actually maybe not since it's at least one of

Reply:
Yeah, could add `oneof`, but I feel like it just makes it more complicated... Maybe I should though, to be clear? ...Yeah, I should probably add it.

Reply:
Wouldn't `oneof` restrict our ability to search by both? Or is it a bit redundant to allow that since we'd only get one community matching this ID when we use this option?

Reply:
Yeah, the idea is that these are both "geo restrictions".

I've clarified this in the pb